core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1331,7 +1332,7 @@ class DAGScheduler(
 
           // TODO: mark the executor as failed only if there were lots of fetch failures on it
           if (bmAddress != null) {
-            handleExecutorLost(bmAddress.executorId, filesLost = true, Some(task.epoch))
+            handleExecutorLost(bmAddress.executorId, slaveLost = true, Some(task.epoch))
```

Comment:
I'm not sure it's correct to assume that a FetchFailure means that all of the executors on the slave were lost.  You could have a failure because one executor died, but the other executors on the host are OK, right?